### PR TITLE
Radio component

### PR DESCRIPTION
**Pull request by kujtimiihoxha**
- Add radio input
- Add radio group
- Add radio button group

## Note
Dependent on #23

### CODE REVIEW

File: src/components/radio/__tests__/Radio.spec.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  });
+
+  test("should trigger mouseenter when over the label", () => {
+    const { container, component } = render(Radio);
+    const onMouseOver = jest.fn();
+    component.$on("mouseenter", onMouseOver);
+    const label = container.getElementsByTagName("LABEL")[0];
+    const evt = document.createEvent("HTMLEvents");
+    evt.initEvent("mouseenter", false, true);
+    label.dispatchEvent(evt);
```
Comment (asaloff) [inline src/components/radio/__tests__/Radio.spec.js:53]: You could also use `fireEvent` here https://testing-library.com/docs/dom-testing-library/api-events

This is fine though. I've just been using that in my tests.

File: src/components/radio/__tests__/RadioGroup.spec.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+      ]
+    });
+    const inputs = container.getElementsByTagName("INPUT");
+
+    expect(inputs.length).toEqual(2);
+    expect(container.innerHTML).toContain("Radio 1");
+    expect(container.innerHTML).toContain("Radio 2");
+  });
+
+  test("should render with sting options", () => {
```
Comment (asaloff) [inline src/components/radio/__tests__/RadioGroup.spec.js:32]:
```suggestion
  test("should render with string options", () => {
```

File: src/components/radio/__tests__/RadioGroup.spec.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+          {
+            component: Radio,
+            options: { value: "radio1" },
+            renderResults: ({ container }) => {
+              radioContainer1 = container;
+            }
+          },
+          { component: Radio, options: { value: "radio2" } }
+        ]
+      }
```
Comment (asaloff) [inline src/components/radio/__tests__/RadioGroup.spec.js:73]: Really cool that we can do this.

File: src/components/radio/Radio.svelte [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  const { getPrefixCls, direction } = $config;
+  prefixCls = prefixCls || getPrefixCls("radio");
+
+  let context = getContext("radioGroupContext") || writable(null);
+
+  let radioProps;
+  let childSlot;
+  let wrapperClassString;
+
+  $: radioProps = { ...$$restProps };
```
[the remark this answers is not in the document]
Comment (kujtimiihoxha) [reply src/components/radio/Radio.svelte:55]: Yeah this is just so users can pass additional props to the base components

File: src/components/radio/Radio.svelte [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+  $: wrapperClassString = classNames(classObj, {
+    [`${prefixCls}-wrapper`]: true,
+    [`${prefixCls}-wrapper-checked`]: radioProps.checked,
+    [`${prefixCls}-wrapper-disabled`]: radioProps.disabled,
+    [`${prefixCls}-wrapper-rtl`]: direction === "rtl"
+  });
+
+  function onChange({ detail }) {
+    $context && $context.setValue(radioProps.value);
```
[the remark this answers is not in the document]
Comment (kujtimiihoxha) [reply src/components/radio/Radio.svelte:75]: Nah this is calling a function in the store.

File: src/components/radio/__tests__/Radio.spec.js (hunk 1 above)
Comment (kujtimiihoxha) [reply src/components/radio/__tests__/Radio.spec.js:53]: I'll start using this from now on.

**Pull request merged**